ui/packages/components/src/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react';
+
+import { Button } from '../Button';
+import { Popover, PopoverClose, PopoverContent, PopoverTrigger } from '../Popover';
+import { Switch, SwitchLabel, SwitchWrapper } from '../Switch';
+import { combineDayAndTime, formatDayString, formatTimeString } from '../utils/date';
+import { Calendar } from './Calendar';
+import { DateInputButton } from './DateInputButton';
+import { TimeInput } from './TimeInput';
+
+type DatePickerProps = {
+  datePickerDate?: Date;
+  setDatePickerDate: React.Dispatch<React.SetStateAction<Date | undefined>>;
```

Comment:
Does this mean consumers are responsible for maintaining the state? If so, what's the reason for avoiding the typical input props of `defaultValue` and `onChange`?

Reply:
When I select a date without a time and click "Apply", the picker closes but `setDatePickerDate` isn't called